root/programs/MultiPurposeAuthSite/MultiPurposeAuthSite/Controllers/AccountController.cs

```diff
@@ -0,0 +1,2221 @@
+                                                shouldLockout: ASPNETIdentityConfig.UserLockoutEnabledByDefault); // ロックアウト
+
+                        // SignInStatus
+                        switch (result)
+                        {
+                            case SignInStatus.Success:
+                                // サインイン成功
+
+                                // テスト機能でSession["state"]のチェックを止めたので不要になった。
+                                // また、ManageControllerの方はログイン済みアクセスになるので。
+
+                                //if (ASPNETIdentityConfig.IsLockedDownRedirectEndpoint)
+                                //{
+
+                                // AppScan指摘の反映
+                                this.FxSessionAbandon();
+                                // SessionIDの切換にはこのコードが必要である模様。
+                                // https://support.microsoft.com/ja-jp/help/899918/how-and-why-session-ids-are-reused-in-asp-net
+                                Response.Cookies.Add(new HttpCookie("ASP.NET_SessionId", ""));
+                                
+                                //}
+                                //else
+                                //{
+                                //    // テスト段階ではOAuth2のテスト機能に支障が出るのでSessionをクリアしない。
+                                //}
+
+                                // イベント・ログ出力
+                                Logging.MyOperationTrace(string.Format("{0}({1}) did sign in.", user.Id, user.UserName));
+
+                                // Open-Redirect対策
+                                if (!string.IsNullOrEmpty(model.ReturnUrl)
+                                    && GetConfigParameter.GetConfigValue("OAuthAuthorizationServerEndpointsRootURI").IndexOf(model.ReturnUrl) != 1)
+                                {   
+                                    return RedirectToLocal(model.ReturnUrl); 
+                                }
+                                else
+                                {
+                                    return RedirectToAction("Index", "Home");
+                                }
+
+                            case SignInStatus.LockedOut:
+                                // ロックアウト
+                                return View("Lockout");
+
+                            case SignInStatus.RequiresVerification:
+                                // EmailConfirmedとは別の2FAが必要。
+
+                                // 検証を求める（2FAなど）。
+                                return this.RedirectToAction(
+                                    "SendCode", new
+                                    {
+                                        ReturnUrl = model.ReturnUrl,  // 戻り先のURL
+                                        RememberMe = model.RememberMe // アカウント記憶
+                                    });
+
+                            case SignInStatus.Failure:
+                            // サインイン失敗
```

Comment:
@daisukenishino 

This program does nothing when sign-in attempt fails.
Can the user know the reason why sign-in attempt failed?